server/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -636,7 +639,11 @@ protected void serverAcceptedChannel(TcpChannel channel) {
         assert addedOnThisCall : "Channel should only be added to accepted channel set once";
         // Mark the channel init time
         channel.getChannelStats().markAccessed(threadPool.relativeTimeInMillis());
-        channel.addCloseListener(ActionListener.wrap(() -> acceptedChannels.remove(channel)));
+        inboundOpenedChannels.inc();
+        channel.addCloseListener(ActionListener.wrap(() -> {
+            acceptedChannels.remove(channel);
+            inboundClosedChannels.inc();
```

Comment:
One of the two metrics here is redundant since `openedChannels - closedChannels = acceptedChannels.size()`. I think it's more useful to expose the number of currently open channels and either the open or close count instead of keeping track of a redundant counter here, both implementation wise and in terms of usefulness for the consumer of these numbers. That's also what is suggested in the meta https://github.com/elastic/elasticsearch/issues/36127 I think